[TI cc23x0 SoC] Add DMA mode to SPI driver

jpanisbl opened the pull request:
This series adds DMA mode support to SPI driver for the TI cc23x0 SoC.

Datasheet: https://www.ti.com/lit/ds/symlink/cc2340r5.pdf

jpanisbl committed:
drivers: spi: cc23x0: Add support for DMA mode

Two DMA channels are assigned to TX and RX respectively:
- A TX DMA single request is asserted when there is space in the FIFO.
- A RX DMA single request is asserted when data is in the FIFO.

Signed-off-by: Julien Panis <jpanis@baylibre.com>

jpanisbl committed:
dts: arm: ti: cc23x0: Add DMA mode support to SPI module

Add support to use DMA mode with cc23x0 SPI module. This consists in
specifying the DMA channels and peripherals.

Signed-off-by: Julien Panis <jpanis@baylibre.com>

jpanisbl commented:
@tbursztyka @teburd @vaishnavachath Could you please review when you have time for this ?

JarmouniA reviewed (changes requested):
A test case is needed in 1 of the SPI tests (loopback for ex) so this code gets at least built in CI.

jpanisbl committed:
tests: drivers: spi: spi_loopback: Add testcase for cc23x0

Enable DMA for the test.

Signed-off-by: Julien Panis <jpanis@baylibre.com>

jpanisbl committed:
drivers: spi: cc23x0: Add loopback mode option

When loopback mode is enabled, the output of transmit serial shifter
is connected to the input of receive serial shifter internally.

Signed-off-by: Julien Panis <jpanis@baylibre.com>

jpanisbl commented:
> A test case is needed in 1 of the SPI tests (loopback for ex) so this code gets at least built in CI.

@JarmouniA Done (I added 2 new commits: 1 in the driver to handle optionally enable loopback, and 1 for the test).
I also re-requested review from @vaishnavachath @tbursztyka.
Thank you all for your time.

jpanisbl commented:
Hi @vaishnavachath Could you please review this PR ? Nothing has changed in the 2 commits you approved 2 weeks ago. Only 2 new commits were added, and they are now approved by JarmouniA who requested the SPI test.
